tests/lib/rendererCoverageCollector.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Utilities for collection of renderer processes coverage
+ *
+ * prepending and appending necessary sequence elements to the test definitions and
+ * for bootstraping the test application instance.
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+
+var fluid = require("gpii-universal"),
+    gpii = fluid.registerNamespace("gpii"),
+    ipcMain = require("electron").ipcMain,
+    BrowserWindow = require("electron").BrowserWindow;
+
+require("gpii-testem");
+
+
+
+fluid.defaults("gpii.tests.app.rendererCoverageServer", {
+    gradeNames:  "gpii.testem.coverage.express",
+    port: 7003,
+    distributeOptions: {
+        record: "%gpii-app/coverage",
+        target: "{that gpii.testem.coverage.receiver.middleware}.options.coverageDir"
+    },
+    listeners: {
+        "onCreate": { // XXX dev
+            funcName: "console.log",
+            args: ["ROUTER CREATED"]
+        }
+    }
+});
+
+
+
+/**
+ * Sends coverage data from a renderer process to the coverageServer.
+ * This function is sent to renderer processes and run there. Once the `gpii.testem.coverage.server`
+ * returns a response it notifies the main process so that the tests would continue.
+ */
+gpii.tests.app.sendRendererCoverage = function () {
```

Comment:
Why did you end up having to write your own coverage client?  I thought you'd be able to use the built-in one with other configuration options (see my other comment for an example).